cmd/server/main.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-sigs/aws_encryption-provider/cloud"
+	"github.com/kubernetes-sigs/aws_encryption-provider/connection"
+	"github.com/kubernetes-sigs/aws_encryption-provider/plugin"
+	"github.com/kubernetes-sigs/aws_encryption-provider/server"
+)
+
+func main() {
+	var (
+		healthzPath = flag.String("healthz-path", "/healthz", "healthcheck path")
+		healthzPort = flag.String("health-port", ":8080", "healthcheck port")
+		addr        = flag.String("listen", "/tmp/awsencryptionprovider.sock", "GRPC listen address")
+		key         = flag.String("key", "", "AWS KMS Key")
+		region      = flag.String("region", "us-east-1", "AWS Region")
+	)
+
+	flag.Parse()
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+
+	glog.Infof("Healthchecks listening on %s", *healthzPort)
+	glog.Infof("Server listening at %s", *addr)
+
+	c, err := cloud.New(*region)
+	if err != nil {
+		glog.Fatalf("Failed to create new KMS service: %v", err)
+	}
+
+	s := server.New()
+	p := plugin.New(*key, c)
+
+	p.Register(s.Server)
+
+	conn, err := connection.New(*addr)
+	if err != nil {
+		glog.Fatalf("Failed to create connection: %v", err)
+	}
+
+	client := p.NewClient(conn)
+
+	go func() {
```

Comment:
Is this a common pattern for grpc services? It certainly makes health-checking more convenient, but I wonder if it would be cleaner to instead bundle a separate script that runs out of process.

Reply:
I think this is a fine place to start.  A script that runs out of process is also a good option - maybe even use that client command.